app/cdap/components/PipelineHistory/PipelineHistoryTableRow.tsx

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { MyPipelineApi } from 'api/pipeline';
+import PipelineModeless from 'components/PipelineDetails/PipelineModeless';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import styled from 'styled-components';
+import T from 'i18n-react';
+import PipelineDetailStore from 'components/PipelineDetails/store';
+import { map } from 'rxjs/operators';
+
+interface IPipelineHistoryTableRowProps {
+  pipelineName: string;
+  appVersion: string;
+}
+
+const PREFIX = 'features.PipelineHistory.table';
+
+export const PipelineHistoryTableRow = ({
+  pipelineName,
+  appVersion,
+}: IPipelineHistoryTableRowProps) => {
+  const namespace = getCurrentNamespace();
+  const pipelineLink = window.getHydratorUrl({
```

Comment:
can you copy getHydratorUrl over into a utility function and not use this one on the window.